packages/components/src/internal/components/settings/DeleteProjectModal.tsx

```diff
@@ -0,0 +1,208 @@
+    const [isDeleting, setIsDeleting] = useState<boolean>(false);
+
+    const { api } = useAppContext();
+    const { container, user } = useServerContext();
+
+    useEffect(() => {
+        (async () => {
+            try {
+                const allSummaries = await api.security.getDeletionSummaries();
+                setSummaries(allSummaries);
+            } catch (e) {
+                // getDeletionSummaries() handles error logging
+                // Purposely leave loading spinner indefinitely in case of summary retrieval failure
+            }
+        })();
+    }, [api.security]);
+
+    const onHide = useCallback(() => {
+        if (!isDeleting) onCancel();
+    }, [isDeleting, onCancel]);
+
+    const onChangeComment = useCallback(e => {
+        setComment(e.target.value);
+    }, []);
+
+    const onDeleteProject = useCallback(async () => {
+        setIsDeleting(true);
+        try {
+            await api.security.deleteContainer(comment);
+
+            const successMsg = `${projectName} successfully deleted.`;
+            const adminProjectsHref = createProductUrl(
+                getPrimaryAppProperties()?.productId,
+                getCurrentAppProperties()?.productId,
+                AppURL.create('admin', 'projects').addParam('successMsg', successMsg).toHref(),
+                container.parentPath
+            ).toString();
+
+            const homeHref = createProductUrl(
+                getPrimaryAppProperties()?.productId,
+                getCurrentAppProperties()?.productId,
+                AppURL.create('home').addParam('successMsg', successMsg).toHref(),
+                container.parentPath
+            ).toString();
+
+            window.location.href = user.isAdmin ? adminProjectsHref : homeHref;
+        } catch (e) {
+            onError(resolveErrorMessage(e) ?? `${projectName} could not be deleted. Please try again.`);
+        }
+    }, [api.security, comment, container.parentPath, onError, projectName, user.isAdmin]);
+
+    const totalCountFromSummaries = useMemo(
+        () =>
+            summaries?.reduce((prev, curr) => {
+                return prev + curr.count;
+            }, 0) ?? 0,
+        [summaries]
+    );
+
+    const noData = useMemo(() => summaries?.length === 0, [summaries]);
+
+    return (
+        <Modal onHide={onHide} show>
+            <Modal.Header closeButton={!isDeleting}>
+                <Modal.Title>{isDeleting ? 'Deleting project' : <> Permanently delete {projectName}? </>} </Modal.Title>
+            </Modal.Header>
+
+            {isDeleting ? (
+                <BodyDeleting totalCountFromSummaries={totalCountFromSummaries} />
+            ) : noData ? (
+                <BodyEmpty onChangeComment={onChangeComment} comment={comment} />
+            ) : (
+                <Body comment={comment} onChangeComment={onChangeComment} summaries={summaries} />
+            )}
+
+            {!isDeleting && (
```

Comment:
Normally we disable the buttons rather than have them disappear. Additionally, we sometimes change the confirmation button text at this time and put a loading spinner if it is something that could be long-running.

Reply:
Probably would be good to go over this kind of stuff at the figma stage—in my head the paradigm was "we're moving to a _new_ deletion-progress modal rather than _replacing_ content from the deletion-form modal" so the button removal made sense, but I'd be happy to change this! And I think the progress bar is supposed to be the loading indicator, if I'm reading correctly where you're saying the loading spinner would be

Reply:
Now that I see the designs in action it makes sense what is happening here. You can disavow this comment.